### ISSUE TITLE
fix put when primaryMaskFormat "7 ([999])", if we use put("788", textfield) and if string first char is 7 mask will remove it, output: "7 88"

### ISSUE THREAD

**Issue by margulan97** [RedMadRobot/input-mask-ios#104]
### Prerequisites

* [x] Put an `X` between the brackets on this line if you have done all of the following:
  * read our [Wiki](https://github.com/RedMadRobot/input-mask-ios/wiki);
  * read the entire [Known issues](https://github.com/RedMadRobot/input-mask-ios#knownissues) section;
  * checked that my issue isn't already [filled](https://github.com/RedMadRobot/input-mask-ios/issues?q=is%3Aissue);
  * searched StackOverflow's [`input-mask`](https://stackoverflow.com/questions/tagged/input-mask) tag for similar problems.

### Describe the bug
if we have any number before mask that is the same as puted Number:
for example: primaryMask "6 [999]", put("699", textfield) output = "6 99" and mask not full. Correct output = "6 699"

**Actual behaviour**
What actually happens.

**Screenshots**

**Platform information**
 - Library version: [last version]

**Additional context**
I try to add in missing digit at first index in string but it was crutch solution, second I try to fix logic in Mask 
public func apply(toText text: CaretString) -> Result
but after fix I have many logic issues with mask. do you have any ideas how to fix it

**Comment by taflanidi** [comment]
Hello @margulan97!
Thanks for your report. 

That's an expected behaviour. Acts the same as if you paste text into the empty field.

**Comment by taflanidi** [comment]
> put("999", textfield) output = "6 999"

Yep, this is also a correct behaviour. When you paste `999` from clipboard, you'd expect all of your digits to be utilised, so the library doesn't cut anything but puts a prefix before the string.

> you don't know which part of the code I need to look at to fix the bug?

Could you please elaborate a little bit more on where the `699` value comes from? The solution would rather depend on it.

For instance, if `699` is a documented value which comes from your backend APIs, and if you know it's always three digits, then the problem is obviously a "frontend rendering" issue; the value will always lack the necessary `6` prefix, thus it's okay to insert it everywhere like you did.

If `699` is stored mobile-wise, I'd rather change the mask format to `{6} [999]` and store whole 4-digit values with prefixes.

Long story short, the meaning behind this prefix dictates the solution.

**Comment by taflanidi** [comment]
So basically the meaning behind the value returned from your backend is "phone number without the country code".

Thus, the country code itself is just a UI thing, and you've got to prefix _all_ the values that are coming from your backend with `7`; the library will do the rest.

**Comment by taflanidi** [comment]
Feel free to reopen should you have any additional questions.
